src/openrct2-ui/windows/ServerStart.cpp

```diff
@@ -47,291 +47,274 @@ enum {
-static void WindowServerStartMouseup(rct_window *w, rct_widgetindex widgetIndex);
-static void WindowServerStartUpdate(rct_window *w);
-static void WindowServerStartTextinput(rct_window *w, rct_widgetindex widgetIndex, char *text);
-static void WindowServerStartInvalidate(rct_window *w);
-static void WindowServerStartPaint(rct_window *w, rct_drawpixelinfo *dpi);
-
-static rct_window_event_list window_server_start_events([](auto& events)
-{
-    events.close = &WindowServerStartClose;
-    events.mouse_up = &WindowServerStartMouseup;
-    events.update = &WindowServerStartUpdate;
-    events.text_input = &WindowServerStartTextinput;
-    events.invalidate = &WindowServerStartInvalidate;
-    events.paint = &WindowServerStartPaint;
-});
-// clang-format on
-
-rct_window* WindowServerStartOpen()
-{
-    rct_window* window;
-
-    // Check if window is already open
-    window = window_bring_to_front_by_class(WC_SERVER_START);
-    if (window != nullptr)
-        return window;
-
-    window = WindowCreateCentred(WW, WH, &window_server_start_events, WC_SERVER_START, WF_10);
-
-    window_server_start_widgets[WIDX_PORT_INPUT].string = _port;
-    window_server_start_widgets[WIDX_NAME_INPUT].string = _name;
-    window_server_start_widgets[WIDX_DESCRIPTION_INPUT].string = _description;
-    window_server_start_widgets[WIDX_GREETING_INPUT].string = _greeting;
-    window_server_start_widgets[WIDX_PASSWORD_INPUT].string = _password;
-    window->widgets = window_server_start_widgets;
-    WindowInitScrollWidgets(window);
-    window->no_list_items = 0;
-    window->selected_list_item = -1;
-    window->frame_no = 0;
-    window->min_width = window->width;
-    window->min_height = window->height;
-    window->max_width = window->min_width;
-    window->max_height = window->min_height;
-
-    window->page = 0;
-    window->list_information_type = 0;
-
-    snprintf(_port, 7, "%u", gConfigNetwork.default_port);
-    safe_strcpy(_name, gConfigNetwork.server_name.c_str(), sizeof(_name));
-    safe_strcpy(_description, gConfigNetwork.server_description.c_str(), sizeof(_description));
-    safe_strcpy(_greeting, gConfigNetwork.server_greeting.c_str(), sizeof(_greeting));
-
-    return window;
-}
-
-static void WindowServerStartClose(rct_window* w)
-{
-}
-
-static void WindowServerStartScenarioselectCallback(const utf8* path)
+class ServerStartWindow final : public Window
 {
-    game_notify_map_change();
-    if (GetContext()->LoadParkFromFile(path, false, true))
+public:
+    void OnOpen() override
     {
-        network_begin_server(gConfigNetwork.default_port, gConfigNetwork.listen_address);
+        _windowServerStartWidgets[WIDX_PORT_INPUT].string = _port;
+        _windowServerStartWidgets[WIDX_NAME_INPUT].string = _name;
+        _windowServerStartWidgets[WIDX_DESCRIPTION_INPUT].string = _description;
+        _windowServerStartWidgets[WIDX_GREETING_INPUT].string = _greeting;
+        _windowServerStartWidgets[WIDX_PASSWORD_INPUT].string = _password;
```

Comment:
I know this is a change but could you move this below `widgets = _windowServerStartWidgets;`. Then you can change `_windowServerStartWidgets[xxx]` to `widgets[xxx]` to conform more with how the rest of the windows work.